sycl/test-e2e/bindless_images/array/read_write_unsampled_array.cpp

```diff
@@ -0,0 +1,457 @@
+// REQUIRES: linux
+// REQUIRES: cuda
+
+// RUN: %clangxx -fsycl -fsycl-targets=%{sycl_triple} %s -o %t.out
+// RUN: %t.out
+
+#include "../bindless_helpers.hpp"
+#include <iostream>
+#include <random>
+#include <sycl/sycl.hpp>
+#include <type_traits>
+
+static sycl::device dev;
+
+// Uncomment to print additional test information
+// #define VERBOSE_PRINT
```

Comment:
```suggestion
#define VERBOSE_PRINT 0
```

and update the checks accordingly. Or (even better) switch to `inline constexpr bool` and `if constexpr`.